app/src/main/java/com/example/scoutconcordia/MapInfoClasses/ShuttleInfo.java

```diff
@@ -526,28 +535,32 @@ public double getEstimatedRouteTimeFromLoyola() {
 
     // These methods will handle the retrieval of times from external encrypted txt files,
     // And then convert them into a comprehensible String array which can be used by this class for calculations.
+    @RequiresApi(api = Build.VERSION_CODES.KITKAT)
     private String[] retrieveMonToThursLoyola() {
 
         ArrayList<String> shuttleArrayList = new ArrayList<>();
 
             try
             {
-                InputStream is = context.getResources().openRawResource(raw.schedule_montothurs_loyola);
-                InputStreamReader readInput = new InputStreamReader(is);
-                StringBuilder sb = new StringBuilder();
-                BufferedReader bfr = new BufferedReader(readInput);
+                try (InputStream is = context.getResources().openRawResource(raw.schedule_montothurs_loyola)) {
+                    try (InputStreamReader readInput = new InputStreamReader(is)) {
+                        StringBuilder sb = new StringBuilder();
+                        try (BufferedReader bfr = new BufferedReader(readInput)) {
```

Comment:
Why are there nested try statements? Is this good practice?

Reply:
This was a naive way of addressing SonarQube's raised issues, however I agree that one Try block is enough. This has been reverted.

Reply:
With only one try block, are the SonarQube issues still resolved?

Multiple try blocks is sometimes the only way, so it depends on the situation. Can you confirm with @JenYn that the SonarQube issues are still resolved on this branch?

Reply:
@vicmanea Since the entire pull request only adds one line of code, I think the SonarQube issues have not been resolved because you reverted them.

It's not a big deal since refactoring and optimization can be done in a future PR.